core/src/crypto/SECP256k1Point.cpp

```diff
@@ -87,6 +87,8 @@ namespace ledger {
             auto flag = secp256k1_ec_pubkey_tweak_add(_context, pubKey, num.data());
             if (flag == 0) throw Exception(api::ErrorCode::RUNTIME_ERROR, "SECP256k1Point SECP256k1Point::generatorMultiply(const std::vector<uint8_t> &n) failed");
```

Comment:
We also need to delete this key before throwing.

Reply:
Why not using a unique_ptr to get free cleanup ?

Reply:
I think we even can have the object on stack.

Reply:
`unique_ptr` has issues here because of opaque types; we would need to provide custom deleters and I think the scope of that function is small enough.

@teams2ua we cannot put them on the stack because they’re opaque types.